src/main/java/pl/joegreen/sergeants/simulator/Tile.java

```diff
@@ -0,0 +1,46 @@
+package pl.joegreen.sergeants.simulator;
+
+import java.util.Optional;
+
+interface Tile {
+
+    int TILE_EMPTY = -1;
```

Comment:
I looked at it again and it does not make sense. Second part of mapdiff is either player index or terrain type which is handled by Tile#getTerrain. It could return FieldTerrainType but then you would still need to handle player index somewhere else. If you want to refactor this you are welcome. That (obviously) goes for any code. :)